converter.go

```diff
@@ -762,3 +763,306 @@ func higherPrecisionEnabled(ctx context.Context) bool {
 	d, ok := v.(bool)
 	return ok && d
 }
+
+func arrowToRecord(record array.Record, rowType []execResponseRowType) (array.Record, error) {
+	s, err := recordToSchema(record.Schema(), rowType)
+	if err != nil {
+		return nil, err
+	}
+
+	var cols []array.Interface
+	numRows := record.NumRows()
+	pool := memory.NewGoAllocator()
+
+	for i, col := range record.Columns() {
```

Comment:
probably gonna need more time looking at this but you I think you have too many `converted = true` statements that could be done outside of `if .. else` for `arrow.DECIMAL`